app/ui/browser-modern/sagas/external-sagas.js

```diff
@@ -40,10 +41,16 @@ function* maximizeWindow() {
 }
 
 function* closeWindow() {
+  // Wait for any additional operations to finish before closing the window.
+  // Add your own here if needed.
+  yield take(EffectTypes.NOTIFY_BROWSER_WINDOW_APP_STATE_DESTROYED);
```

Comment:
Is this a situation where you want 'before-close-browser-window' and then 'close-browser-window'?  Can a window _deny_ it's closing with this change set?  (I think it can't.)  Can a window do some potentially expensive IO operation to prepare it's saved state before closing?  (I think it can.)